tests/bridge/test_bridge.py

```diff
@@ -65,6 +66,105 @@
+    badger = connect_badger(badger_config.prod_json)
+    bridge = connect_bridge(badger, badger_config.prod_json)
+    swap = connect_swap(badger_config.prod_json)
+    _upgrade_bridge(badger, bridge)
+    _upgrade_swap(badger, swap)
+    _deploy_bridge_mocks(badger, bridge)
+
+    slippage = .03
+    amount = 1 * 10**8
+
+    v = vault["address"]
+    if v == AddressZero:
+        v = MockVault.deploy(
+            vault["id"],
+            vault["symbol"],
+            vault["token"],
+            {"from": badger.deployer}
+        ).address
+        # Must approve mock vaults to mint/burn to/from.
+        bridge.adapter.setVaultApproval(
+            v,
+            True,
+            {"from": badger.devMultisig},
+        )
+    else:
+        badger.sett_system.vaults[vault["id"]].approveContractAccess(
+            bridge.adapter,
+            {"from": badger.devMultisig},
+        )
+
+    if poolId == 3:
+        v = wbtcAddr
+
+    #Setting contract addresses in adapter
+    bridge.adapter.setVaultPoolId(v, poolId, {"from": badger.devMultisig})
+    bridge.adapter.setIbbtcContracts(ibbtcContract, peakContract, wbtcPeakContract, {"from": badger.devMultisig})
+
+    #governance / guestlist / contract approval stuff
+    gov = interface.ICore(coreContract).owner()
+    interface.ICore(coreContract).setGuestList(AddressZero, {"from": gov})
+
+    gov2 = interface.IBadgerSettPeak(peakContract).owner()
+    interface.IBadgerSettPeak(peakContract).approveContractAccess(bridge.adapter, {"from": gov2})
+
+    gov3 = interface.IBadgerYearnWbtcPeak(wbtcPeakContract).owner()
+    interface.IBadgerYearnWbtcPeak(wbtcPeakContract).approveContractAccess(bridge.adapter, {"from": gov3})
+
+    #allowance approval
+    #interface.IERC20(v).approve(bridge.adapter, amount, {"from": accounts[0]})
+    #if vault["symbol"] == "bwBTC":
+    #    interface.IERC20(wbtcAddr).approve(wbtcPeakContract, amount, {"from": bridge.adapter})
+    #else:
+    #    interface.IERC20(v).approve(peakContract, amount, {"from": bridge.adapter})
+
+    #minting
+    accountsBalanceBefore = interface.IERC20(ibbtcContract).balanceOf(accounts[0].address)
+    bridge.adapter.mint(
+        vault["inToken"],
+        slippage * 10**4,
+        accounts[0],
+        v,
+        amount,
+        # Darknode args hash/sig optional since gateway is mocked.
+        "",
+        "", 
+        True,
+        {"from": accounts[0]},
+    )
+    accountsBalanceAfter = interface.IERC20(ibbtcContract).balanceOf(accounts[0].address)
+
+    assert accountsBalanceAfter > accountsBalanceBefore
+
+    #burning
+    interface.IERC20(ibbtcContract).approve(bridge.adapter, accountsBalanceAfter, {"from": accounts[0]})
+    bridge.adapter.burn(
+        ibbtcContract,
+        v,
+        slippage * 10**4,
+        accounts[0].address,
+        accountsBalanceAfter,
+        {"from": accounts[0]},
+    )
+
+    assert interface.IERC20(ibbtcContract).balanceOf(accounts[0].address) == 0
+
```

Comment:
Probably want to check the balance of the mock gateway contract to make sure that we actually "burned" the tokens (seems to be not happening based on early `return` in the contract code). 

The mock gateway will transfer the tokens back to itself (the real gateway actually burns the tokens to 0x0 address but it's fine if we test this way).